SDL/Hit.cuh

```diff
@@ -91,7 +85,7 @@ namespace SDL
       return x - n * float(2.f * float(M_PI));
     }
     CUDA_HOSTDEV inline float phi(float x, float y) {
-      return phi_mpi_pi(float(M_PI) + ATan2(-y, -x));
+      return atan2f(y,x);
```

Comment:
is it safe to avoid nans? Most of `ATan2` implementation was meant to regularize the result if inputs are 0

Reply:
I removed it because the Alpaka branch removes it and still runs fine. I think that the atan2f function by C++ considers the case on x=0 internally, so we don't need to do it extra. Except that if we put (x,y)=(0,0), then atan2 function will give issues. But I think we don't have this in our case.

Reply:
I think the usage of wrap it by a phi_mpi_pi function is to constrain the output angle into the range of (-pi,pi)
But I googled the usage of the atan function and the default output is (-pi, pi), so I think another protection is not necessary

Reply:
I think the usage of wrap it by a phi_mpi_pi function is to constrain the output angle into the range of (-pi,pi)
But I googled the usage of the atan function and the default output is (-pi, pi), so I think another protection is not necessary

Reply:
As the messages above your last replies say, the whole discussion is about the ATan2 usage and the x=0, y=0 case. Given this,  I am confused about how the (-π,π) range fits into that discussion (maybe I missed some comment?).

But, in any case, if you are sure that `atan2` will work for the (0,0) case, fine by me.

Reply:
I checked that the atan2 function on the (0,0) case, and found that it returns pi. so I added a protection on the phi_mpi_pi function. Thank you for insisting!